crates/pathfinder/src/state.rs

```diff
@@ -217,39 +467,44 @@ async fn update<T: Transport>(
 
     // Download additional block information from sequencer. Use a custom timeout with retry strategy
     // to work-around the sequencer's poor performance (spurious lack of response) for early blocks.
-    let block = loop {
-        match sequencer
-            .block_by_number_with_timeout(
-                BlockNumberOrTag::Number(update_log.block_number),
-                Duration::from_secs(3),
-            )
-            .await
-        {
-            Ok(block) => break block,
-            Err(err) => {
-                use sequencer::error::*;
-                match err {
-                    SequencerError::TransportError(terr) if terr.is_timeout() => continue,
-                    other => {
-                        let err = anyhow::anyhow!("{}", other)
-                            .context("Downloading StarkNet block from sequencer");
-                        return Err(UpdateError::Other(err));
-                    }
-                }
-            }
-        }
-    };
+    // let block = loop {
+    //     match sequencer
+    //         .block_by_number_with_timeout(
+    //             BlockNumberOrTag::Number(update_log.block_number),
+    //             Duration::from_secs(3),
+    //         )
+    //         .await
+    //     {
+    //         Ok(block) => break block,
+    //         Err(err) => {
+    //             use sequencer::error::*;
+    //             match err {
+    //                 SequencerError::TransportError(terr) if terr.is_timeout() => continue,
+    //                 other => {
+    //                     let err = anyhow::anyhow!("{}", other)
+    //                         .context("Downloading StarkNet block from sequencer");
+    //                     return Err(UpdateError::Other(err));
+    //                 }
+    //             }
+    //         }
+    //     }
+    // };
 
     // Verify sequencer root against L1.
-    let sequencer_root = block.state_root.context("Sequencer state root missing")?;
+    // let sequencer_root = block.state_root.context("Sequencer state root missing")?;
 
-    if sequencer_root != update_log.global_root {
-        return Err(UpdateError::Other(anyhow::anyhow!(
-            "Sequencer state root did not match L1."
-        )));
-    }
+    // if sequencer_root != update_log.global_root {
+    //     return Err(UpdateError::Other(anyhow::anyhow!(
+    //         "Sequencer state root did not match L1."
+    //     )));
+    // }
 
-    let block_hash = block.block_hash.context("Sequencer block hash missing")?;
+    // let block_hash = block.block_hash.context("Sequencer block hash missing")?;
+    let block_hash = StarknetBlockHash(pedersen_hash(
+        update_log.global_root.0,
+        StarkHash::from_be_slice(&update_log.block_number.0.to_be_bytes()).unwrap(),
+    ));
```

Comment:
This is obviously horrible and shouldn't be left.

Reply:
But I'm thinking since this is the one we use for testing, it should be the one we just have commited.

Reply:
Left because it's how it's done before L2 is responding.